index.js

```diff
@@ -343,6 +343,64 @@ function Physics (mcData, world) {
     }
   }
 
+  function getLookingVector (entity) {
+    // given a yaw pitch, we need the looking vector
+
+    // yaw is right handed rotation about y (up) starting from -z (north)
+    // pitch is -90 looking down, 90 looking up, 0 looking at horizon
+    // lets get its coordinate system.
+    // let x' = -z (north)
+    // let y' = -x (west)
+    // let z' = y (up)
+
+    // the non normalized looking vector in x', y', z' space is
+    // x' is sin(yaw)
+    // y' is cos(yaw)
+    // z' is tan(pitch)
+
+    // substituting back in x, y, z, we get the looking vector in the normal x, y, z space
+    // -z = sin(yaw) => z = -sin(yaw)
+    // -x = cos(yaw) => x = -cos(yaw)
+    // y = tan(pitch)
+
+    // normalizing the vectors, we divide each by |sqrt(x*x + y*y + z*z)|
+    // x*x + y*y = sin^2 + cos^2 = 1
+    // so |sqrt(xx+yy+zz)| = |sqrt(1+tan^2(pitch))|
+    //     = |sqrt(1+sin^2(pitch)/cos^2(pitch))|
+    //     = |sqrt((cos^2+sin^2)/cos^2(pitch))|
+    //     = |sqrt(1/cos^2(pitch))|
+    //     = |+/- 1/cos(pitch)|
+    //     = 1/cos(pitch) since pitch in [-90, 90]
+
+    // the looking vector is therefore
+    // x = -cos(yaw) * cos(pitch)
+    // y = tan(pitch) * cos(pitch) = sin(pitch)
+    // z = -sin(yaw) * cos(pitch)
```

Comment:
You seem to have flipped the values of x and z in this comment (implementation itself looks fine).

Reply:
oh strange, yeah, x and z are backwards

Reply:
I think the code is right since I also cross checked it against the lookAt function
```js
    const yaw = Math.atan2(-delta.x, -delta.z)
    const groundDistance = Math.sqrt(delta.x * delta.x + delta.z * delta.z)
    const pitch = Math.atan2(delta.y, groundDistance)
```
`yaw = atan2(-x, -z)` implies `-x = sin(yaw)` and `-z = cos(yaw)` by definition of atan2

Also, the definition of sin is the y component of a point on the unit circle, so it should be `y' = sin(yaw)` not `y' = cos(yaw)`

Reply:
ok comment should be fixed, and I added unit tests to confirm that it was flying in a direction consistent with the lookAt function's definition of the coordinates. Side note, is there a place that the coordinate frame definitions are noted? I still don't know what a notchean angle is. Also, pitch seems to be upside down in the f3 menu, but not here

Reply:
See [here](https://github.com/PrismarineJS/mineflayer/blob/master/lib/conversions.js) for conversions between notchian angles and radians/degrees